src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -204,12 +205,22 @@ private[snowflake] class JDBCWrapper {
             throw new IllegalArgumentException(s"Don't know how to save $field of type ${field.name} to Snowflake")
         }
         val nullable = if (field.nullable) "" else "NOT NULL"
-        sb.append(s""", ${name.replace("\"", "\\\"")} $typ $nullable""".trim)
+        sb.append(s""", ${if (isQuoted(name)) name else quotedName(name)} $typ $nullable""".trim)
```

Comment:
Yes, the escaping of the quotes is necessary for Snowflake to correctly parse the queries. Let's maybe change the above to something like:

`val formattedIdentifierName = if (isQuoted(name) name else quotedName(name)
  sb.append(s""",formattedName.replace(..........),.....)`